examples/platforms/kw41z/radio.c

```diff
@@ -0,0 +1,900 @@
+ *
+ */
+
+#include <stdint.h>
+#include <string.h>
+#include "fsl_device_registers.h"
+#include "openthread-core-kw41z-config.h"
+#include "fsl_xcvr.h"
+#include "openthread/platform/radio.h"
+#include "openthread/platform/diag.h"
+#include <utils/code_utils.h>
+
+#define DOUBLE_BUFFERING            (1)
+#define DEFAULT_CHANNEL             (11)
+#define IEEE802154_ACK_REQUEST      (1 << 5)
+#define DEFAULT_CCA_MODE            (XCVR_CCA_MODE1_c)
+#define IEEE802154_TURNAROUND_LEN   (12)
+#define IEEE802154_CCA_LEN          (8)
+#define IEEE802154_PHY_SHR_LEN      (10)
+#define IEEE802154_ACK_WAIT         (54)
+#define ZLL_IRQSTS_TMR_ALL_MSK_MASK (ZLL_IRQSTS_TMR1MSK_MASK | \
+                                     ZLL_IRQSTS_TMR2MSK_MASK | \
+                                     ZLL_IRQSTS_TMR3MSK_MASK | \
+                                     ZLL_IRQSTS_TMR4MSK_MASK )
+
+typedef enum xcvr_state_tag
+{
+    XCVR_Idle_c,
+    XCVR_RX_c,
+    XCVR_TX_c,
+    XCVR_CCA_c,
+    XCVR_TR_c,
+    XCVR_CCCA_c,
+} xcvr_state_t;
+
+typedef enum xcvr_cca_type_tag
+{
+    XCVR_ED_c,            /* energy detect - CCA bit not active, not to be used for T and CCCA sequences */
+    XCVR_CCA_MODE1_c,     /* energy detect - CCA bit ACTIVE */
+    SCVR_CCA_MODE2_c,     /* 802.15.4 compliant signal detect - CCA bit ACTIVE */
+    XCVR_CCA_MODE3_c      /* 802.15.4 compliant signal detect and energy detect - CCA bit ACTIVE */
+} xcvr_cca_type_t;
+
+static PhyState     sState = kStateDisabled;
+static uint16_t     sPanId;
+static uint8_t      sExtSrcAddrBitmap[(RADIO_CONFIG_SRC_MATCH_ENTRY_NUM + 7) / 8];
+static uint8_t      sChannel;
+static int8_t       sMaxED;
+static int8_t       sAutoTxPwrLevel = 0;
+
+/* ISR Signaling Flags */
+static bool         sTxDone     = false;
+static bool         sRxDone     = false;
+static bool         sEdScanDone = false;
+static bool         sAckFpState;
+static ThreadError  sTxStatus;
+
+static RadioPacket  sTxPacket;
+static RadioPacket  sRxPacket;
+#if DOUBLE_BUFFERING
+static uint8_t      sRxData[kMaxPHYPacketSize];
+#endif
+
+/* Private functions */
+static void         rf_abort(void);
+static xcvr_state_t rf_get_state(void);
+static void         rf_set_channel(uint8_t channel);
+static void         rf_set_tx_power(int8_t tx_power);
+static uint8_t      rf_lqi_adjust(uint8_t hwLqi);
+static int8_t       rf_lqi_to_rssi(uint8_t lqi);
+static uint32_t     rf_get_timestamp(void);
+static void         rf_set_timeout(uint32_t abs_timeout);
+static uint16_t     rf_get_addr_checksum(uint8_t *pAddr, bool ExtendedAddr, uint16_t PanId);
+static ThreadError  rf_add_addr_table_entry(uint16_t checksum, bool extendedAddr);
+static ThreadError  rf_remove_addr_table_entry(uint16_t checksum);
+static ThreadError  rf_remove_addr_table_entry_index(uint8_t index);
+
+PhyState otPlatRadioGetState(otInstance *aInstance)
+{
+    (void)aInstance;
+    return sState;
+}
+
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
```

Comment:
When I try with one kw41z board, seems this function will cause reset the board. Could you have a check here?